posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,168 @@
+                           
+[source]
+----
+ibmcloud login --apikey $IBM_CLOUD_API_KEY -g $IBM_CLOUD_RSGRP
+ibmcloud ks cluster config --cluster $CLUSTER-ID
+kubectl config current-context
+kubectl create -f deploy/deployment.yaml #( simple k8s deployment command )
+kubectl create -f deploy/service.yaml #( simple k8s service creation command )
+kubectl create -f deploy/route.yaml #( simple k8s route creation command )
+----
+
+Make sure that your Kubernetes configuration files are stored in the same git repository as your Jenkinsfile in a sub-directory called `deploy`. Also ensure that the Docker image name in the Kubernetes deployment configuration file is updated according to the container image name/tag in the Dockerfile (manually, or programmatically if it needs to change at run time):
+
+[.imageblock.img_border_light]
+image::/img/blog/deploymentyaml.png[Image reference in deployment yaml,width=40%,align="center"]
+
+When Jenkins has checked out the Java application code for the code build, all the Kubernetes configuration files are also downloaded to the Jenkins workspace so that Jenkins can run the IBM Cloud and Kubernetes commands to connect to the Kubernetes cluster and deploy the application.
+
+See the link:https://kubernetes.io/docs/reference/kubectl/cheatsheet/[Kubernetes documentation] for other commands.
+
+== QA testing cloud-native Java applications with Jenkins
+
+Apart from running JUnit test cases along with the code build phase, Jenkins can trigger functional and integration QA test cases automatically after deploying the cloud-native Java application.
+
+Configure the test cases in the Jenkins job and test it manually. Create a remote job identifier authentication token in the "Trigger builds remotely" section under "Build Triggers". Trigger this test case from the Docker "entrypoint" file by using a remote rest API call that uses this authentication token as the identifier.
+
+For example, run the following command in a terminal: 
+
+[source]
+----
+curl -I -u <auth-token> https://<jenkins-host>/job/<job-name>/build?token=<remote-job-identifier-authentication-token>
+----
+
+You can generate an authentication token (auth-token) with link:https://www.postman.com[Postman] using the Jenkins login credentials.
+
+== Kubernetes monitoring tools
+
+Kubernetes provides commands to check the application or cluster logs and memory and CPU usage through the commands like: 
+
+[source]
+----
+kubectl logs ..
+cat /sys/fs/cgroup/cpu/cpuacct.usage (after connecting to k8s pod)
+cat /sys/fs/cgroup/memory/memory.usage_in_bytes (after connecting to k8s pod)
+----
+
+There are several applications available that you can integrate with Kubernetes to persist logs and usage statistics, such as link:https://grafana.com/oss/loki/[Grafana] and link:https://prometheus.io[Prometheus]. 
```

Comment:
Let us make a reference to Liberty's support for these tools too?

Reply:
Done